### ISSUE TITLE
Request for upgrade to WebSocket returned bad response

### ISSUE THREAD

**Issue by RobertLarsen** [processone/ejabberd#566]
Request:
```
GET /ws-xmpp HTTP/1.1
Origin: http://www.komogvind.dk
Sec-WebSocket-Protocol: xmpp
Sec-WebSocket-Key: +cLaJc229FHZxEdesnpWow==
Connection: Upgrade
Upgrade: Websocket
Sec-WebSocket-Version: 13
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
Host: chat.cego.dk:8888
DNT: 1
Cache-Control: no-cache
```

Response:
```
HTTP/1.1 200 OK
Content-Length: 506
Content-Type: text/xml; charset=utf-8
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: Content-Type

<?xml version='1.0'?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml'><head><title>ejabberd ejabberd_http_ws</title></head><body><h1>ejabberd ejabberd_http_ws</h1><p>An implementation of <a href='http://tools.ietf.org/html/rfc6455'>WebSocket protocol</a></p><p>This web page is only informative. To use WebSocket connection you need a Jabber/XMPP client that supports it.</p></body></html>
```

To the best of my knowlege that was a valid upgrade request. WebSockets have been enabled and is working.
I'm using the source from Git and the last commit was 31e356c126cd5b967df9e7b560ce63f2b3b1529b.

**Comment by prefiks** [comment]
Hello,

I think that your problem will be fixed if you use build from newer git commit, i think your problem will be fixed by 0cd77ab942d563363d0a54f786e4747ef160c384, and your build was created from tree not having it.

Please try building from 15.04 tag at least as there was couple important fixes to websocket in that release.

**Comment by RobertLarsen** [comment]
@prefiks Absolutely right, thx
